### PR TITLE
Modernize Python 2 code

### PR DESCRIPTION
**Pull request by cclauss**
Use the __futurize__ tool to do automated conversions.

### CODE REVIEW

**Comment by AetherUnbound** [comment]
@cclauss Thanks for your work to help us convert to python 3! Do you mind ensuring that the build passes before I merge this in?

**Comment by cclauss** [comment]
Yes.  I got caught up in other things today but will come back with fixes soon.  We definitely do not want to megrge while Python 2.7 tests are not passing.

**Comment by cclauss** [comment]
This last modification permits pip install on Python 3.6.3 because that is Travis CI's current Python 3.6.  This allows us to test only on Travis without opening up full support on all versions of Python 3.

**Comment by cclauss** [comment]
Ready for review.  This PR touches a lot of files so the chances of conflicts are high.

File: cellprofiler/gui/figure.py [hunk 1]
```diff
@@ -912,14 +912,15 @@ def show_imshow_popup_menu(self, pos, subplot_xy):
         popup = self.get_imshow_menu(subplot_xy)
         self.PopupMenu(popup, pos)
 
-    def get_imshow_menu(self, (x, y)):
+    def get_imshow_menu(self, xxx_todo_changeme):
```
Comment (bengerman13) [inline cellprofiler/gui/figure.py]: is this ready to be merged?

File: cellprofiler/modules/editobjectsmanually.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # coding=utf-8
 
-import _help
+from __future__ import absolute_import
+# coding=utf-8
```
Comment (bengerman13) [inline cellprofiler/modules/editobjectsmanually.py]: doubled `coding` line

File: cellprofiler/modules/expandorshrinkobjects.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # coding=utf-8
 
-import _help
+from __future__ import absolute_import
+# coding=utf-8
```
Comment (bengerman13) [inline cellprofiler/modules/expandorshrinkobjects.py]: Doubled `coding`

File: cellprofiler/modules/filterobjects.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # coding=utf-8
 
-import _help
+from __future__ import absolute_import
+# coding=utf-8
```
Comment (bengerman13) [inline cellprofiler/modules/filterobjects.py]: doubled `coding` line

File: cellprofiler/modules/identifyobjectsingrid.py [hunk 5]
```diff
@@ -1,6 +1,9 @@
 # coding=utf-8
 
-import _help
+from __future__ import absolute_import
+# coding=utf-8
```
Comment (bengerman13) [inline cellprofiler/modules/identifyobjectsingrid.py]: duplicated `coding` line

File: cellprofiler/modules/identifyobjectsmanually.py [hunk 6]
```diff
@@ -1,6 +1,9 @@
 # coding=utf-8
 
-import _help
+from __future__ import absolute_import
+# coding=utf-8
```
Comment (bengerman13) [inline cellprofiler/modules/identifyobjectsmanually.py]: duplicated `coding` line

File: cellprofiler/modules/images.py [hunk 7]
```diff
@@ -390,7 +390,7 @@ def __init__(self):
                                                              'directory', "Directory", self.fn_filter,
                                                              predicates, doc="Apply the rule to directories")
 
-    def fn_filter(self, (node_type, modpath, module), *args):
+    def fn_filter(self, xxx_todo_changeme, *args):
```
Comment (bengerman13) [inline cellprofiler/modules/images.py]: does this name need to be changed?
same on 430, 502, 584

File: cellprofiler/modules/measureobjectintensitydistribution.py [hunk 8]
```diff
@@ -1,6 +1,9 @@
 # coding=utf-8
 
-import _help
+from __future__ import absolute_import
+# coding=utf-8
```
Comment (bengerman13) [inline cellprofiler/modules/measureobjectintensitydistribution.py]: duplicated `coding` line

File: cellprofiler/gui/figure.py (hunk 1 above)
Comment (bengerman13) [inline cellprofiler/gui/figure.py]: does this name need to be changed?

**Comment by cclauss** [comment]
Good catch.  Thanks!  Fixed.

File: cellprofiler/modules/images.py [hunk 9]
```diff
@@ -390,7 +390,7 @@ def __init__(self):
                                                              'directory', "Directory", self.fn_filter,
                                                              predicates, doc="Apply the rule to directories")
 
-    def fn_filter(self, (node_type, modpath, module), *args):
+    def fn_filter(self, node_type__modpath__module, *args):
```
Comment (AetherUnbound) [inline cellprofiler/modules/images.py]: This doesn't sit well with me...but neither does tuple-ized arguments!

File: cellprofiler/pipeline.py [hunk 10]
```diff
@@ -1,6 +1,7 @@
 """Pipeline.py - an ordered set of modules to be executed
 """
 from __future__ import with_statement
+from __future__ import print_function
```
Comment (AetherUnbound) [inline cellprofiler/pipeline.py]: These two lines can probably be condensed

**Comment by AetherUnbound** [comment]
I think this should be ready to go when these tests pass!

**Pull request merged**